src/main/java/net/snowflake/client/jdbc/SnowflakeBaseResultSet.java

```diff
@@ -1347,11 +1361,83 @@ public <T> T getObject(int columnIndex, Class<T> type) throws SQLException {
       SQLInput sqlInput = (SQLInput) getObject(columnIndex);
       instance.readSQL(sqlInput, null);
       return (T) instance;
+    } else if (Map.class.isAssignableFrom(type)) {
+      JsonNode jsonNode = ((JsonSqlInput) getObject(columnIndex)).getInput();
+      return (T) OBJECT_MAPPER.convertValue(jsonNode, new TypeReference<Map<String, Object>>() {});
     } else {
       return (T) getObject(columnIndex);
     }
   }
 
+  public <T> List<T> getList(int columnIndex, Class<T> type) throws SQLException {
+    T[] sqlInputs = getArray(columnIndex, type);
+    return Arrays.asList(sqlInputs);
+  }
+
+  public <T> T[] getArray(int columnIndex, Class<T> type) throws SQLException {
+    Optional<Supplier<SQLData>> typeFactory = SnowflakeObjectTypeFactories.get(type);
+    Map[] sqlInputs = (Map[]) getArray(columnIndex).getArray();
+    T[] arr = (T[]) java.lang.reflect.Array.newInstance(type, sqlInputs.length);
+    AtomicInteger counter = new AtomicInteger(0);
+    Arrays.stream(sqlInputs)
+        .map(
+            map ->
+                new JsonSqlInput(
+                    OBJECT_MAPPER.convertValue(map, JsonNode.class),
+                    session,
+                    sfBaseResultSet.getConverters(),
+                    sfBaseResultSet
+                        .getMetaData()
+                        .getColumnMetadata()
+                        .get(columnIndex - 1)
+                        .getFields()))
+        .forEach(
+            i -> {
+              SQLData instance =
+                  typeFactory
+                      .map(Supplier::get)
+                      .orElseGet(() -> createUsingReflection((Class<SQLData>) type));
+              try {
+                instance.readSQL(i, null);
+              } catch (SQLException e) {
+                throw new RuntimeException(e);
```

Comment:
I don't believe we should remap this exceptions here. We should just propagate the previous one.

Reply:
fixed